src/cmds/fs/ls.c

```diff
@@ -168,27 +168,40 @@ int main(int argc, char **argv) {
 	if (optind < argc) {
 		struct stat sb;
 
-		if (-1 == stat(argv[optind], &sb)) {
-			return -errno;
-		}
+		optind--;
 
-		if (!S_ISDIR(sb.st_mode)) {
-			printer(argv[optind], &sb);
-			return 0;
-		}
+		while (optind++ < argc - 1) {
```

Comment:
Why do we need `optind--` here? optind should point to the first file already. Probably we can use `do .. while` here?

Reply:
I also feel strange but I didn't remember for a while. After I modified all of them， do I need make a another PR. Sorry,  I'm not familiar with this

Reply:
No, you should push your new changes to this one